### PR TITLE
(feat) Next iteration of all xml-support

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

**Why**

For background and context, see https://github.com/vespa-engine/pyvespa/pull/915

**What** 

Added some more meat needed to ensure compatibility with old approach. 
Still a good way to go, but getting closer. 
Next iteration will have tests of compatibility with old approach, and more tests for new approach.

### CODE REVIEW

File: vespa/configuration/vt.py [hunk 1]
```diff
@@ -64,6 +65,7 @@ def __iter__(self):
 
 
 def attrmap(o):
+    o = dict(_global="global").get(o, o)
```
Comment (glebashnik) [inline vespa/configuration/vt.py:68]: What is the purpose of this?

File: vespa/configuration/vt.py [hunk 2]
```diff
@@ -98,16 +100,21 @@ def vt(
     **kw,
 ):
     "Create an `VT` structure for `to_xml()`"
-    return VT(
-        tag.lower(), *_preproc(c, kw, attrmap=attrmap, valmap=valmap), void_=void_
-    )
+    # NB! fastcore.xml uses tag.lower() for tag names. This is not done here.
+    return VT(tag, *_preproc(c, kw, attrmap=attrmap, valmap=valmap), void_=void_)
```
Comment (glebashnik) [inline vespa/configuration/vt.py:104]: What does preproc do? 
A docstring for preproc would be nice.

File: vespa/configuration/vt.py [hunk 3]
```diff
@@ -157,6 +164,8 @@ def _to_xml(elm, lvl, indent, do_escape):
 
     # Handle void (self-closing) tags
     if elm.void_:
+        if isinstance(elm.void_, str):
```
Comment (glebashnik) [inline vespa/configuration/vt.py:167]: For contxt, void_ is a bool?
Why check for str?

**Review by glebashnik — APPROVED** [review]
Looks good but can be hard to understand with doc strings and type hints.
Consider adding them for non-trivial code.
See specific questions.

**Comment by thomasht86** [comment]
Thanks a lot for the input. Will address in following PR as discussed.

**Pull request merged**